subgraphs/curve-finance/src/common/types.ts

```diff
@@ -28,6 +26,51 @@ export class RewardsInfoType {
   }
 }
 
+export class RewardData {
+  private _rewardToken: Address;
+  private _rewardRate: BigInt;
+  private _periodFinish: BigInt;
+
+  constructor(gaugeAddress: Address, rewardToken: Address) {
+    const gaugeContract = LiquidityGaugeContract.bind(gaugeAddress);
+
+    const rewardDataV1 = gaugeContract.try_reward_data(rewardToken);
+    if (rewardDataV1.reverted) {
```

Comment:
If contract calls revert do we want to log anything?

Reply:
No, as there are only three types of reward contracts, it will go through the if else condition and find the correct one.